SSA/WellTypedFramework.lean

```diff
@@ -1,10 +1,14 @@
 import SSA.Framework
 import Mathlib.Data.Option.Basic
 
-class BaseTypeClass (BaseType : Type)  : Type 1 where
+/-
+Typeclass for a `baseType` which is a godel code of 
```

Comment:
I'm happy with having a better name than `BaseTypeClass` for this! Honestly, almost any name does it. I'm curious why Gödel though? It's not quite the same as the Gödel encoding of statements as numbers, which is the only analogy that fits here I know of.

Reply:
Ah, as a generalization of godel coding, where we reflect the metatheory (PA) into the theory using natural numbers. I've seen this type of thing be called `Code`, though I am unsure where --- either in ATAPL, or in one of Andrej Bauer's code.